### ISSUE TITLE
"No Rows found" and "Loading..." should not be displayed at the same time

### ISSUE THREAD

**Issue by bernalrs** [tannerlinsley/react-table#189]
## Problem Description
I would like to open up this issue for discussions. I have noticed that when I have an empty table displaying "No rows found" and then I trigger the table to populate while setting the loading prop to true this creates an overlay where both texts are visible. 

I would like to suggest replacing the "No Row found " text with loading or alternative only display the loading overlay but not both at the same time. @tannerlinsley  what do you think ?

## Code Snippet(s)

`            <ReactTable
                loading= {this.props.loading}
                data={this.props.data}
                columns={columns}
                className="-striped -highlight"
                pivotBy={['P1', 'P2', 'P3']}
                showPagination={false}
            />`

## Steps to Reproduce
1. list the steps 

1. Render an empty table 
2. set loading to true

## System Information
* **Browser & Browser Version:**
* **Node Version:**
* **OS Version:**
* **NPM Version:**
* **Yarn Version:**
![empty table](https://cloud.githubusercontent.com/assets/245996/24910330/4d5eee44-1ebf-11e7-96ab-9c3e5b9b93fb.PNG)

**Comment by tannerlinsley** [comment]
There was a small discussion in the slack channel about only allowing `noData` text to be displayed when and if there were no rows and the table was loading.  This would have been a very tightly integrated solution for this specific use case, but would significantly lock down the API for other users who may want the opposite. 

We decided to wait for now to see if a similar use case appears, and in the mean time rely on user to dictate around this special circumstance using the api to override `noData` text to be empty`.

**Comment by ghost23** [comment]
I just stumbled upon the same issue. Unfortunately for me, it is not as easy as setting the text to being empty, because there is still an empty white box slightly visible behind the loading overlay.

I would also like to add, that I can not imagine, how anyone would want both messages to appear at the same time.

Anyway, a compromise would be, if I would be able to specify my own render method for noData. This way, I would be able to render null, if data is loading.

**Comment by aaronschwartz** [comment]
@ghost23 You should be able to completely override the component that is rendered. 

Try setting the `NoDataComponent` on the table.

**Comment by ghost23** [comment]
Yes, I guess, I will have to do it that way.

**Comment by bernalrs** [comment]
Seems this is now fixed in version 6.

**Comment by andrei-sheina** [comment]
@aaronschwartz @ghost23 Correct me if I wrong, but an overriding component, that will render on render null, based on isLoding will not work, since NoDataComponent will be mounted right when ReactTable will be mounted (before async loading even gets started).
@ghost23 did manage to fix this issue?

**Comment by aaronschwartz** [comment]
@andrei-sheina  I don't really understand what your goal is.

If you're taking full control of the `loading` and `NoDataComponent` props you're in control of when everything gets rendered. 

In that case you know exactly when your data is being loaded or not. 

You could try something like this:
```
loading={this.state.loading}
NoDataComponent={
  this.state.finishedLoadingData && !this.state.loading ? (<div>No Data</div>) : null
}
```

If you have something else in mind it would be great if you could recreate it in codepen.

**Comment by andrei-sheina** [comment]
@aaronschwartz ok, sorry for misleading. Overriding NoDataComponent actually works perfectly. It was just a wrong conditional statement in my code.
In case anyone else will run into this issue, here is working solution:

```
ComponentWithDataToRenderInTable.js

<ReactTable
            className='-striped -highlight'
            data={state.data}
            columns={columns}
            loading={state.isLoading}
            LoadingComponent={CustomLoadingComponent}
            NoDataComponent={CustomNoDataComponent}
          />

CustomNoDataComponent.js

const CustomNoDataComponent = ({ state, ...rest }) => {
  if (state.isLoading) {
    return null;
  }
  return <div className="rt-noData">No rows found</div>
}

export default CustomNoDataComponent;
```

**Comment by crtvhd** [comment]
Well, we should be lucky the authors of `react-table` decided to give us that much flexibility 😎
Here's my take on above-described problem:

TLDR; Don't render `NoDataComponent` as long as `loading: true`.

```es6

// NOTE: Map `loading` into props a NoDataComponent will receive
// so we can hide the component as long as data fetching is in progress
const getNoDataProps = (props) => ({
  loading: props.loading,
});

[...]

<ReactTable
  getNoDataProps={getNoDataProps}
  NoDataComponent={NoDataComponent}
/>

[...]

const NoDataComponent = (props) => {
  const { children, loading } = props;

  return loading
  ? null
  : (
    <styled.NoData>
      {children}
    </styled.NoData>
  );
};
```

**Comment by kaidoj** [comment]
I did it like this 
```js

<ReactTable
  noDataText={!this.state.loading?'No rows found':''}
  NoDataComponent={NoDataComponent}
/>

onst NoDataComponent = (props) => {
    const {
        children
    } = props;

    return children === '' ?
        null :
        (
          <div className="rt-noData">
             {children}
         </div>
        );
};

```
